packages/react-components/react-storybook-addon/src/docs/CopyAsMarkdownButton.tsx

```diff
@@ -0,0 +1,228 @@
+    if (!targetWindow) {
+      return;
+    }
+
+    // Create new AbortController for this request
+    const abortController = new AbortController();
+    abortControllerRef.current = abortController;
+
+    // Show loading toast that persists until updated
+    dispatchToast(
+      <Toast>
+        <ToastTitle media={<Spinner />}>Copying page content...</ToastTitle>
+      </Toast>,
+      {
+        toastId,
+        intent: 'info',
+        timeout: -1, // Never auto-dismiss
+      },
+    );
+
+    try {
+      // Use cached content if available, otherwise fetch from API
+      if (!markdownContentCache.current) {
+        markdownContentCache.current = await fetchMarkdownContent(targetWindow, markdownUrl, abortController.signal);
+      }
+
+      // Copy to clipboard
+      await targetWindow?.navigator.clipboard.writeText(markdownContentCache.current);
+
+      // Update toast to success
+      updateToast({
+        content: (
+          <Toast>
+            <ToastTitle>Page content copied to clipboard!</ToastTitle>
+          </Toast>
+        ),
+        intent: 'success',
+        toastId,
+        timeout: 3000,
+      });
+    } catch (error) {
+      // Don't show error if request was aborted
+      if (abortController.signal.aborted) {
+        return;
+      }
+
+      const errorMessage = error instanceof Error ? error.message : String(error);
+
+      // Update toast to error
+      updateToast({
+        content: (
+          <Toast>
+            <ToastTitle>Failed to copy page content: {errorMessage}</ToastTitle>
+          </Toast>
+        ),
+        intent: 'error',
+        toastId,
+        timeout: 3000,
+      });
+    } finally {
+      // Clear the abort controller ref to allow new requests
+      abortControllerRef.current = null;
+    }
+  }, [dispatchToast, updateToast, toastId, markdownUrl, targetWindow]);
+
+  /** Opens the markdown content in a new browser tab */
+  const openInNewTab = React.useCallback(() => {
+    targetWindow?.open(markdownUrl, '_blank');
+  }, [markdownUrl, targetWindow]);
+
+  if (!storyId) {
+    return null;
+  }
+
+  return (
+    <>
+      <Menu positioning="below-end">
+        <MenuTrigger disableButtonEnhancement>
+          {(triggerProps: MenuButtonProps) => (
+            <SplitButton
+              className={styles.button}
+              menuButton={triggerProps}
+              primaryActionButton={{
+                appearance: 'secondary',
+                icon: <MarkdownIcon />,
+                onClick: copyPageContentToClipboard,
+                'aria-label': 'Copy page content as markdown to clipboard',
+              }}
+              aria-label="Copy page as markdown"
+            >
+              Copy Page
+            </SplitButton>
+          )}
+        </MenuTrigger>
+
+        <MenuPopover>
+          <MenuList>
+            <MenuItem icon={<MarkdownIcon />} onClick={openInNewTab}>
+              View as Markdown
```

Comment:
why not making this as primary action ? 

I'm missing the usefulness of the workflow behind copying MD to clipboard manually. can you provide more context on that one pls ?